src/main/java/minicraft/item/ArcaneFertilizerItem.java

```diff
@@ -0,0 +1,53 @@
+package minicraft.item;
+
+import minicraft.entity.Direction;
+import minicraft.entity.mob.Player;
+import minicraft.entity.particle.Particle;
+import minicraft.gfx.SpriteLinker;
+import minicraft.level.Level;
+import minicraft.level.tile.BoostablePlantTile;
+import minicraft.level.tile.Tile;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.ArrayList;
+import java.util.Random;
+
+public class ArcaneFertilizerItem extends StackableItem {
+	public static ArrayList<Item> getAllInstances() {
+		ArrayList<Item> items = new ArrayList<>();
+		items.add(new ArcaneFertilizerItem("Arcane Fertilizer", new SpriteLinker.LinkedSprite(SpriteLinker.SpriteType.Item, "arcane_fertilizer")));
+		return items;
+	}
+
+	protected ArcaneFertilizerItem(String name, SpriteLinker.LinkedSprite sprite) {
+		super(name, sprite);
+	}
+
+	private static final SpriteLinker.LinkedSprite particleSprite = new SpriteLinker.LinkedSprite(SpriteLinker.SpriteType.Entity, "glint");
+
+	@Override
+	public boolean interactOn(Tile tile, Level level, int xt, int yt, Player player, Direction attackDir) {
+		if (tile instanceof BoostablePlantTile) {
+			if (((BoostablePlantTile) tile).isValidBoostablePlantTarget(level, xt, yt)) {
+				if (((BoostablePlantTile) tile).isPlantBoostSuccess(level, xt, yt)) {
+					((BoostablePlantTile) tile).performPlantBoost(level, xt, yt);
+				}
```

Comment:
The first one checks if the target tile is a valid type; the second one checks if the tile can be boosted according to its current state.

Reply:
It is quite unusual to do this:

	if (((BoostablePlant) tile).isPlantBoostSuccess(level, xt, yt)) {
		((BoostablePlant) tile).performPlantBoost(level, xt, yt);
	}

Normally this would just be one function call, and the checking would be done by the tile.
As a general rule of thumb: don't call low-level code from high-level functions.